crates/client/block_production/src/lib.rs

```diff
@@ -473,10 +489,19 @@ impl<Mempool: MempoolProvider> BlockProductionTask<Mempool> {
                         continue
                     }
 
-                    if let Err(err) = self.on_pending_time_tick() {
-                        tracing::error!("Pending block update task has errored: {err:#}");
+                    match self.on_pending_time_tick().await {
+                        Ok(block_closed) => {
+                            if block_closed {
+                                interval_pending_block_update.reset_at(instant + interval_pending_block_update.period());
+                                interval_block_time.reset_at(instant + interval_block_time.period());
+                            }
+                            self.current_pending_tick += 1;
```

Comment:
could you reset the current_pending_tick to 0 as well?
(i believe the current pending tick variable is now useless, is that the case?

Reply:
yeah I thought about it as well, pushing the changes that I think made sense, lmk wdyt